packages/shared/src/hooks/useSquadNavigation.ts

```diff
@@ -1,39 +1,42 @@
 import { useRouter } from 'next/router';
-import { useEffect, useMemo } from 'react';
-import { LazyModal } from '../components/modals/common/types';
-import { NewSquadModalProps } from '../components/modals/NewSquadModal';
+import { useCallback, useContext, useEffect } from 'react';
 import { Origin } from '../lib/analytics';
-import { useLazyModal } from './useLazyModal';
 import usePersistentContext from './usePersistentContext';
+import AuthContext from '../contexts/AuthContext';
+import FeaturesContext from '../contexts/FeaturesContext';
 
-type ModalProps = Omit<NewSquadModalProps, 'onRequestClose' | 'isOpen'>;
-interface UseCreateSquadModal {
-  openNewSquadModal: (props?: ModalProps) => void;
+type OpenNewSquadProps = { origin: Origin };
+type EditSquadProps = { handle: string };
+interface UseSquadNavigation {
+  openNewSquad: (props?: OpenNewSquadProps) => void;
+  editSquad: (props: EditSquadProps) => void;
 }
 
 const SQUAD_ONBOARDING = 'hasTriedSquadOnboarding';
 const SQUAD_INVITE_PATHNAME = '/squads/[handle]/[token]';
 
-type UseCreateSquadModalProps = {
-  hasSquads: boolean;
-  hasAccess: boolean;
-  isFlagsFetched: boolean;
-};
-
-export const useCreateSquadModal = ({
-  hasSquads = false,
-  hasAccess = false,
-  isFlagsFetched,
-}: UseCreateSquadModalProps): UseCreateSquadModal => {
+export const useSquadNavigation = (): UseSquadNavigation => {
```

Comment:
Is this not a bit of overkill.
Basically we just have hrefs to static pages now right, with no additional logic.

Feels like we could simply use `href={newSquad}` and editSquad on each respective place.
Do note we are removing the auto-open function here, so maybe you can incorporate that?
https://dailydotdev.atlassian.net/jira/software/projects/WT/boards/11?selectedIssue=WT-1527